### PR TITLE
Add addons:rename command

### PR DESCRIPTION
**Pull request by mnoble**
Adds a command to rename an add-on.

@heroku/add-ons 
@dickeyxxx

### CODE REVIEW

File: commands/addons/rename.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+let cli = require('heroku-cli-util');
+let co  = require('co');
+
+let run = cli.command(function (ctx, api) {
+    return co(function* () {
+        let addon    = yield api.get(`/addons/${ctx.args.addon}`);
+        let addonUrl = `/apps/${addon.app.id}/addons/${ctx.args.addon}`;
```
Comment (bjeanes) [inline commands/addons/rename.js]: I'd use `addon.uuid` here for where we are `POST`ing just to be sure.

File: commands/addons/rename.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        cli.log(`${oldName} successfully renamed to ${newName}.`);
+    });
+});
+
+let topic = 'addons';
+
+module.exports = {
+    topic:       topic,
+    command:     'rename',
+    wantsApp:    false,
```
Comment (bjeanes) [inline commands/addons/rename.js]: We might want this to be true so that you can do `heroku addons:rename -a app DATABASE my-new-name`

**Comment by bjeanes** [comment]
What do you think about adding a test for either or both of: trying to rename to a name that is already taken or renaming to a name with invalid chars. Both are simply API errors, so they could be tested generically as just getting a 422 or whatever from API, I think.

**Comment by bjeanes** [comment]
Just a few minor comments, other than that a huge :+1: and thanks for picking this up!

File: commands/addons/rename.js (hunk 1 above)
Comment (mnoble) [inline commands/addons/rename.js]: Yea, good point.

**Comment by dickeyxxx** [comment]
code lgtm. I don't think a different test is needed for those other cases since the implementation doesn't need to account for it in either case. If it was doing some kind of client-side validation I would say it would, but in this case it's just a failed api request that's caught by heroku-cli-util and shows a message, so I think it's good with just the one API failure case

File: commands/addons/rename.js (hunk 2 above)
Comment (mnoble) [inline commands/addons/rename.js]: That command makes me sad, but you're right.

**Comment by mnoble** [comment]
@bjeanes Yea, what @dickeyxxx said. I presume handling API errors correctly is tested in the CLI.

**Comment by bjeanes** [comment]
@mnoble @dickeyxxx SGTM then :)

**Pull request merged**